popcorn/lib/popcorn.rb

```diff
@@ -0,0 +1,45 @@
+require "popcorn/version"
+require "popcorn/popcorn_graph"
+require "popcorn/node"
+
+module Popcorn
+  include PopcornGraph
+
+  VOCABULARY = %w(pop popcorn corn cop no poop porno)
+
+  def self.find_all
+    words = Set.new
+
+    NODES.each do |node|
+      (words += find_by_node(node)) if node
+    end
+
+    words
+  end
+
+  def self.find_by_node node, str = '', words = Set.new
```

Comment:
This method does too much.
It would be better to have some generic `Graph#traverse` method that accepts a block and yields a node to the caller. The block could return `true` or `false` back to indicate that `traverse` should not go deeper in case it's an invalid prefix.
This way it'd separate graph traversal from word detection logic